test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+					Skip(fmt.Sprintf("designated worker node %q has SMT level %d - minimum required 2", workerRTNode.Name, smtLevel))
+				}
+				// create a deployment to deploy gu pods
+				dp := newDeployment()
+				testNode := make(map[string]string)
+				testNode["kubernetes.io/hostname"] = workerRTNode.Name
+				dp.Spec.Template.Spec.NodeSelector = testNode
+				err := testclient.Client.Create(context.TODO(), dp)
+				Expect(err).ToNot(HaveOccurred(), "Unable to create Deployment")
+
+				defer func() {
+					// delete deployment
+					testlog.Infof("Deleting Deployment %v", dp.Name)
+					err := testclient.Client.Delete(context.TODO(), dp)
+					Expect(err).ToNot(HaveOccurred())
+				}()
+
+				// Check the cpus of ovn kube node pods and ovs services
+				ovnContainerCpus, err := getOvnContainerCpus(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to fetch cpus of the containers inside ovn kubenode pods")
+				testlog.Infof("cpus used by ovn kube node pods %v", ovnContainerCpus)
+				pidList, err := getOVSServicesPid(workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+				for _, pid := range pidList {
+					cmd := []string{"taskset", "-pc", pid}
+					cpumask, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+					Expect(err).ToNot(HaveOccurred(), "failed to fetch cpus of %s", pid)
+					mask := strings.SplitAfter(cpumask, " ")
+					maskSet, err := cpuset.Parse(mask[len(mask)-1])
+					testlog.Infof("cpus used by ovs services are: %s", maskSet.String())
+					Expect(ovnContainerCpus).To(Equal(maskSet.String()), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", ovnContainerCpus, maskSet.String())
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+
+				testlog.Info("Rebooting the node")
+				// reboot the node, for that we change the numa policy to best-effort
+				// Note: this is used only to trigger reboot
+				policy := "best-effort"
+				// Need to make some changes to pp , causing system reboot
+				// and check if activation files is modified or deleted
+				profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+				Expect(err).ToNot(HaveOccurred(), "Unable to fetch latest performance profile")
+				currentPolicy := profile.Spec.NUMA.TopologyPolicy
+				if *currentPolicy == "best-effort" {
+					policy = "restricted"
+				}
+				profile.Spec.NUMA = &performancev2.NUMA{
+					TopologyPolicy: &policy,
+				}
+
+				By("Updating the performance profile")
+				profiles.UpdateWithRetry(profile)
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+				By("Waiting for MCP being updated")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+				// Check the cpus of ovn kube node pods and ovs services
+				ovnContainerCpus, err = getOvnContainerCpus(workerRTNode)
```

Comment:
This is racy on many levels:
- The reboot killed the guaranteed pods and the Deployment will take its time to restart them
- The cpu manager takes time to apply the cpu masks

Reply:
I am addressing this by checking first the deployment is created pods and the pods are in running state. before we do any checks further.